File: src/main/java/com/leijendary/spring/boot/template/core/model/UUIDModel.java
package com.leijendary.spring.boot.template.core.model;

import java.util.UUID;

import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@MappedSuperclass
public abstract class UUIDModel extends AppModel {

    @Id
    @GeneratedValue
    private UUID id;
}

File: src/main/kotlin/com/leijendary/spring/boot/template/core/model/UUIDModel.kt
package com.leijendary.spring.boot.template.core.model

import java.util.*
import java.util.UUID.randomUUID
import javax.persistence.Column
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
abstract class UUIDModel {
    @Id
    @GeneratedValue
    var id: UUID = randomUUID()

    @Column(insertable = false)
    var rowId: Long? = null
}